kavm/src/kavm/__main__.py

```diff
@@ -111,23 +118,40 @@ def exec_run(
         raise RuntimeError('Cannot access KAVM_LIB environment variable. Is it set?')
     kavm_lib_dir = Path(str(os.environ.get('KAVM_LIB')))
 
-    if not teal_programs_parser:
-        teal_programs_parser = kavm_lib_dir / 'scripts/parse-teal-programs.sh'
+    if not teal_parser:
+        teal_parser = definition_dir / 'parser_TealInputPgm_TEAL-PARSER-SYNTAX'
     if not avm_simulation_parser:
         avm_simulation_parser = kavm_lib_dir / 'scripts/parse-avm-simulation.sh'
+    if not avm_json_parser:
+        avm_json_parser = definition_dir / 'parser_JSON_AVM-TESTING-SYNTAX'
     try:
-        proc_result = kavm.run_avm_simulation(
-            input_file=input_file,
-            output=output,
-            profile=profile,
-            teal_sources_dir=teal_sources_dir,
-            teal_programs_parser=teal_programs_parser,
-            avm_simulation_parser=avm_simulation_parser,
-            depth=depth,
-        )
-        if not output == 'none':
-            print(proc_result.stdout)
-        exit(proc_result.returncode)
+        if input_file.suffix == '.json':
+            scenario = KAVMScenario.from_json(input_file.read_text())
+            with tempfile.TemporaryDirectory() as decompiled_teal_dir:
+                src_files = os.listdir(teal_sources_dir)
+                for file_name in src_files:
+                    full_file_name = os.path.join(teal_sources_dir, file_name)
+                    if os.path.isfile(full_file_name):
+                        shutil.copy(full_file_name, decompiled_teal_dir)
+                scenario.decompile_teal_programs(Path(decompiled_teal_dir))
+                proc_result = kavm.run_avm_json(
+                    scenario=scenario.to_json(),
+                    teals=kavm.parse_teals(scenario._teal_files, Path(decompiled_teal_dir)),
+                    output=output,
+                    profile=profile,
+                    depth=depth if depth else 0,
+                )
+            if not output in ['none', 'final-state-json']:
+                print(proc_result.stdout)
+            if output == 'final-state-json' and proc_result.returncode == 0:
+                print(json.dumps(json.loads(proc_result.stderr), indent=4))
+            if proc_result.returncode:
+                print(proc_result.stdout)
+                print(proc_result.stderr)
+            exit(proc_result.returncode)
+        else:
+            print(f'Unrecognized input file extension: {input_file.suffix}')
+            exit(1)
```

Comment:
Here you can `raise ValueError(f'Unrecognized ...')`, then we get a nice stack trace too.